internal/rbd/rbd_util.go

```diff
@@ -603,36 +603,31 @@ func (rv *rbdVolume) getCloneDepth(ctx context.Context) (uint, error) {
 	vol.Monitors = rv.Monitors
 	vol.RbdImageName = rv.RbdImageName
 	vol.conn = rv.conn.Copy()
-
+	currVol := &vol
 	for {
-		if vol.RbdImageName == "" {
+		if currVol.RbdImageName == "" {
 			return depth, nil
 		}
-		err := vol.openIoctx()
+		err := currVol.openIoctx()
 		if err != nil {
 			return depth, err
 		}
-
-		err = vol.getImageInfo()
-		// FIXME: create and destroy the vol inside the loop.
-		// see https://github.com/ceph/ceph-csi/pull/1838#discussion_r598530807
-		vol.ioctx.Destroy()
-		vol.ioctx = nil
+		parentVol, err := currVol.getParentVolume()
 		if err != nil {
 			// if the parent image is moved to trash the name will be present
 			// in rbd image info but the image will be in trash, in that case
 			// return the found depth
 			if errors.Is(err, ErrImageNotFound) {
 				return depth, nil
 			}
-			util.ErrorLog(ctx, "failed to check depth on image %s: %s", &vol, err)
+			util.ErrorLog(ctx, "failed to check depth on image %s: %s", &currVol, err)
```

Comment:
`currVol` is already a `*rbdVolume`, using `&currVol` will likely print the pointer to `currVol` and not the result of `currVol.String()` as expected